core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ast/Shape.java

```diff
@@ -523,4 +530,17 @@ private static void calculateTargetChain(List<Shape> parsed) {
 		}
 	}
 
+	@Override
+	public String toString() {
+		Model statements = toModel(new DynamicModel(new LinkedHashModelFactory()));
+		statements.setNamespace(SHACL.NS);
+		WriterConfig writerConfig = new WriterConfig();
+		writerConfig.set(BasicWriterSettings.PRETTY_PRINT, true);
+		writerConfig.set(BasicWriterSettings.INLINE_BLANK_NODES, true);
+
+		StringWriter stringWriter = new StringWriter();
+		Rio.write(statements, stringWriter, RDFFormat.TURTLE, writerConfig);
+		return stringWriter.toString().replace("@prefix sh: <http://www.w3.org/ns/shacl#> .", "").trim();
+	}
+
```

Comment:
This is used for logging the shape, which is why I take some liberties with stripping out the namespace line.